src/components/Rover.tsx

```diff
@@ -0,0 +1,17 @@
+import { useParams, Redirect } from 'react-router-dom';
+
+interface RoverParams {
+    rover: string;
+}
+
+export function Rover() {
+    const { rover } = useParams<RoverParams>();
+
+    const regexMatch = /(opportunity)|(spirit)|(curiosity)/i
+
+    return (
```

Comment:
So could we change this to having an earlier return if the `rover` doesn't match the regex. This is so that we can add the more complicated logic that will involve the rover name being correct after this return
```suggestion
    if (!rover.match(regex)) {
        return <Redirect to="/" />
    }
    
    return (
```